scripts/upgrage-legacy-rules.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at: [package root]/LICENSE.txt
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+
+import sys
+import argparse
+import textwrap
+from typing import List, Optional
+from pathlib import Path
+
+import yaml
+
+from capa.main import collect_rule_file_paths
+from capa.rules import Rule
+from capa.features.address import NO_ADDRESS
+
+dynamic_features = ("api", "string", "substring", "number", "description", "regex", "match", "os")
+engine_words = ("and", "or", "optional", "not")
+static_scopes = ("function", "basic block", "instruction")
+dynamic_scopes = ("thread",)
+
+
+def rec_features_list(static, context=False):
+    """
+    takes in a list of static features, and returns it alongside a list of dynamic-only features
+    """
+    dynamic = []
+    for node in static:
+        for key, value in node.items():
+            pass
+        if isinstance(value, list):
+            # is either subscope or ceng
+            if key in (*static_scopes, *dynamic_scopes):
+                # is subscope
+                stat, dyn = rec_scope(key, value, context)
+                if not context and dyn:
+                    dynamic.append({"or": [stat, dyn]})
+                elif context == "d" and dyn:
```

Comment:
i think its totally reasonable to use the strings "static" and "dynamic" rather than going through the work to define enums, etc. and, we can also provide a type hint like `Union[Literal["static"], Literal["dynamic"]]` in the function signature so people know what to expect.

Reply:
I agree. Using "static" and "dynamic" alongside type annotations sounds good to me, I'll add that.